tls/s2n_handshake_io.c

```diff
@@ -742,7 +791,9 @@ static int handshake_read_io(struct s2n_connection *conn)
         conn->in_status = ENCRYPTED;
 
         /* Advance the state machine */
-        GUARD(s2n_advance_message(conn));
+        if (ACTIVE_STATE(conn).record_type == TLS_CHANGE_CIPHER_SPEC) {
```

Comment:
since the code previous didn't have this check, what the cases when this condition is not true?

Reply:
As long as the TLS1.2 handshake is going according to the spec, this condition is always true. However, the TLS1.3 spec specifies that we should be able to handle the change cipher spec messages arriving up at any point in the handshake.